cps2pc

```diff
@@ -0,0 +1,206 @@
+def get_link_features(link_features, platform):
+    _supported_kernels = ['cygwin', 'linux', 'darwin']
+    _supported_link_features = {x: {'threads': '-lpthread'} for x in _supported_kernels}
+    try:
+        features = []
+        for feature in link_features:
+            features.append(_supported_link_features[platform][feature])
+        return features
+    except KeyError as k:
+        raise Exception("Unknown link feature %s on %s" % (k.message, platform))
+
+def get_compile_features(compile_features, compiler):
+    # For loop with catch KeyError exception and check with WError, and W... flags
+    _re_we_features = re.compile('((no)?(warn|error):(error$)?)(.*)')
+    _W_we_prefixes = {'warn:error': '-Werror', 'nowarn:error': '-Wno-error', 'warn:': '-W', 'nowarn:': '-Wno-',
+                      'error:': '-Werror=', 'noerror:': '-Wno-error='}
+    _compilers_we_prefixes = {'gcc': _W_we_prefixes, 'clang': _W_we_prefixes}
+    _supported_compiler_feature_list = ['c99', 'c11', 'c++03', 'c++11', 'c++14', 'c++17', 'gnu']
+    _supported_gcc_features = {x: '--std=%s' % x for x in _supported_compiler_feature_list}
+    _supported_clang_features = _supported_gcc_features
+    _supported_compile_features = {'gcc': _supported_gcc_features, 'clang': _supported_clang_features}
+    features = []
+    for feature in compile_features:
+        try:
+            features.append(_supported_compile_features[compiler][feature])
+        except KeyError as k:
+            ma = _re_we_features.match(feature)
+            if ma is not None:
+                features.append(_compilers_we_prefixes[compiler][ma.group(1)]+ma.group(5))
+                continue
+            raise Exception("Unknown link feature %s on %s" % (k.message, compiler))
+    return features
+
+def get_cpp_runtime_vendor(package):
+    _supported_cpp_runtime_vendors = {'gnu': 'libstdc++', 'llvm': 'libc++'}
+    if hasattr(package, 'platform') and hasattr(package.platform, 'runtime'):
+        try:
+            runtime = _supported_cpp_runtime_vendors[package.platform.runtime.vendor]
+            return ['-l%s' % runtime]
+        except KeyError:
+            raise Exception("Unknown runtime %s" % runtime)
+    return []
+
+def get_definitions_flags(compile_definitions, compiler):
+    _re_definitions = re.compile('(!)?(.*)')
+    _unix_definition_prefixes = {None: '-D', '!': '-U'}
+    _compilers_definition_prefixes = {'gcc': _unix_definition_prefixes, 'clang': _unix_definition_prefixes}
+    definitions = []
+    for definition in compile_definitions:
+        try:
+            ma = _re_definitions.match(definition)  # No need to test result `ma` as regex catches everything.
+            prefix = _compilers_definition_prefixes[compiler][ma.group(1)]
+            definitions.append(prefix + ma.group(2))
+        except:
+            raise Exception("Unknown definition argument %s" % definition)
+    return definitions
+
+
```

Comment:
Second blank line should be separator line instead. (Also occurs several times below.)